docs/src/main/asciidoc/grpc-service-consumption.adoc

```diff
@@ -0,0 +1,208 @@
+@Inject @GrpcService("hello") MutinyGreeterGrpc.MutinyGreeterStub mutinyHelloService;
+
+@GET
+@Path("/blocking/{name}")
+public String helloBlocking(@PathParam("name") String name) {
+    return blockingHelloService.sayHello(HelloRequest.newBuilder().setName(name).build()).getMessage();
+}
+
+@GET
+@Path("/mutiny/{name}")
+public Uni<String> helloMutiny(@PathParam("name") String name) {
+    return mutinyHelloService.sayHello(HelloRequest.newBuilder().setName(name).build())
+            .onItem().apply(HelloReply::getMessage);
+}
+----
+
+Note that in this example, the `hello.host` property must be set.
+
+=== Handling streams
+
+gRPC allows sending and receiving streams:
+
+[source]
+----
+service Streaming {
+    rpc Source(Empty) returns (stream Item) {} // Returns a stream
+    rpc Sink(stream Item) returns (Empty) {}   // Reads a stream
+    rpc Pipe(stream Item) returns (stream Item) {}  // Reads a streams and return a streams
+}
+----
+
+Using the Mutiny stub, you can interact with these as follows:
+
+[source, java]
+----
+package io.quarkus.grpc.example.streaming;
+
+import io.grpc.examples.streaming.Empty;
+import io.grpc.examples.streaming.Item;
+import io.grpc.examples.streaming.MutinyStreamingGrpc;
+import io.quarkus.grpc.runtime.annotations.GrpcService;
+import io.smallrye.mutiny.Multi;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/streaming")
+@Produces(MediaType.APPLICATION_JSON)
+public class StreamingEndpoint {
+
+    @Inject @GrpcService("streaming") MutinyStreamingGrpc.MutinyStreamingStub client;
+
+    @GET
+    public Multi<String> invokeSource() {
+        // Retrieve a stream
+        return client.source(Empty.newBuilder().build())
+                .onItem().apply(Item::getValue);
+    }
+
+    @GET
+    @Path("sink/{max}")
+    public Uni<Void> invokeSink(@PathParam("max") int max) {
+        // Send a stream and wait for completion
+        Multi<Item> inputs = Multi.createFrom().range(0, max)
+                .map(i -> Integer.toString(i))
+                .map(i -> Item.newBuilder().setValue(i).build());
+        return client.sink(inputs).onItem().ignore().andContinueWithNull();
+    }
+
+    @GET
+    @Path("/{max}")
+    public Multi<String> invokePipe(@PathParam("max") int max) {
+        // Send a stream and retrieve a stream
+        Multi<Item> inputs = Multi.createFrom().range(0, max)
+                .map(i -> Integer.toString(i))
+                .map(i -> Item.newBuilder().setValue(i).build());
+        return client.pipe(inputs).onItem().apply(Item::getValue);
+    }
+
+}
+
+----
+
+== Client configuration
+
+IMPORTANT: Client configuration is controlled with the `@GrpcService` annotation.
+
+For a given `$NAME` (value set in the `@GrpcService` annotation) , the following configuration properties are supported:
```

Comment:
OK, I have my answer and I don't like that. We should use proper Quarkus config. First it will avoid conflicting with other things, then it will be automatically documented.

Reply:
Fixed. It's now using a Map<String, GrpcClientConfiguration>